includes/Modules/Sign_In_With_Google.php

```diff
@@ -428,4 +469,90 @@ public function has_placed_tag_in_content( $content ) {
 
 		return Module_Tag_Matchers::NO_TAG_FOUND;
 	}
+
+	/**
+	 * Returns the disconnect URL for the specified user.
+	 *
+	 * @since n.e.x.t
+	 *
+	 * @param int $user_id WordPress User ID.
+	 */
+	public static function disconnect_url( $user_id = null ) {
+		return add_query_arg(
+			array(
+				'action'  => self::DISCONNECT_ACTION,
+				'nonce'   => wp_create_nonce( self::DISCONNECT_ACTION ),
+				'user_id' => $user_id,
+			),
+			admin_url( 'index.php' )
+		);
+	}
+
+	/**
+	 * Handles the disconnect action.
+	 *
+	 * @since n.e.x.t
+	 *
+	 * @param string $nonce Nonce.
+	 */
+	public function handle_disconnect_user( $nonce ) {
+		if ( ! wp_verify_nonce( $nonce, self::DISCONNECT_ACTION ) ) {
+			$authentication = new Authentication( $this->context );
+			$authentication->invalid_nonce_error( self::DISCONNECT_ACTION );
+		}
+
+		if ( ! isset( $_REQUEST['user_id'] ) ) {
+			return;
+		}
+		$user_id = (int) $_REQUEST['user_id'];
+
+		// Only allow this action for admins or users own setting.
+		if ( current_user_can( Permissions::SETUP ) || get_current_user_id() === $user_id ) {
+			delete_user_meta( $user_id, $this->user_options->get_meta_key( Hashed_User_ID::OPTION ) );
+		}
+		wp_safe_redirect( get_edit_user_link( $user_id ) );
+		exit;
+	}
+
+	/**
+	 * Displays a disconnect button on user profile pages.
+	 *
+	 * @since n.e.x.t
+	 *
+	 * @param WP_User $user WordPress user object.
+	 */
+	private function render_disconnect_profile( WP_User $user ) {
+		$current_user_google_id = get_user_meta( $user->ID, $this->user_options->get_meta_key( Hashed_User_ID::OPTION ), true );
+
+		// Don't show if the user does not have a Google ID save in user meta.
+		if ( empty( $current_user_google_id ) ) {
+			return;
+		}
+
+		// Only show to admins or users own settings.
+		if ( ! ( current_user_can( Permissions::SETUP ) || get_current_user_id() === $user->ID ) ) {
+			return;
+		}
+		?>
+<div id="googlesitekit-sign-in-with-google-disconnect">
+	<h2><?php esc_html_e( 'Sign in with Google', 'google-site-kit' ); ?></h2>
+	<p>
+		<?php
+		esc_html_e(
+			'This user can sign in with their Google account.',
+			'google-site-kit'
+		);
+		?>
+	</p>
+	<p>
+		<a
+			class="button button-secondary"
+			href="<?php echo esc_url( self::disconnect_url( $user->ID ) ); ?>"
+		>
+			<?php esc_html_e( 'Disconnect Google Account', 'google-site-kit' ); ?>
+		</a>
```

Comment:
Huh, I see this button as a much more "distinct" action. I don't think a toggle makes sense, and I think it makes sense for the button to be immediate and not require further user interaction, similar to the "Log out everywhere else" button:

![Screenshot 2024-11-15 at 22 06 18](https://github.com/user-attachments/assets/9b8b52ce-783e-40be-b26e-468db472d81a)

I don't think we should be leveraging the same form/hooks here, because this should be a discrete/distinct action.

Reply:
Sure, I guess I should have looked to see how some of these others worked, but after looking, they're all a little different 😄 I do think this example here is a good one which is similar and possible on both your own profile and when editing another user.

This one uses an admin-ajax request which uses the user_id and nonce from elements on the page, which is a bit odd.
<img width="210" alt="image" src="https://github.com/user-attachments/assets/9e0f2336-e329-4155-b454-fd0cdf111e65">

The send password reset also uses admin-ajax but has a user-specific nonce provided via l10n data on the page
<img width="256" alt="image" src="https://github.com/user-attachments/assets/75d2d67d-198f-41dc-9f91-c2f4b2402423">
https://github.com/WordPress/WordPress/blob/ea472597956e31c9aa4421300b13ae481679df07/wp-includes/script-loader.php#L1225-L1228

You can see the requests are quite similar too.

Application passwords however use REST routes and those are very straightforward.

All of them have a certain interactivity about them though as feedback in response to the action.

I'll take a look again and we can see about refactoring this later.